statsd/receiver.go

```diff
@@ -149,24 +151,16 @@ func (mr *MetricReceiver) handleMessage(addr net.Addr, msg []byte) {
 		if len(line) > 1 {
 			metric, err := mr.parseLine(line)
 			if err != nil {
-				log.Errorf("Error parsing line %q from %s: %v", line, addr, err)
+				log.Warnf("Error parsing line %q from %s: %v", line, addr, err)
 				mr.increment("bad_lines_seen", 1)
 				continue
 			}
-			source := strings.Split(addr.String(), ":")
-			hostname := source[0]
-			if net.ParseIP(hostname) != nil {
-				if mr.Cloud != nil {
-					instance, err := cloudprovider.GetInstance(mr.Cloud, hostname)
-					if err != nil {
-						log.Errorf("Error retrieving instance details from cloud provider %s: %v", mr.Cloud.ProviderName(), err)
-					} else {
-						hostname = instance.ID
-						metric.Tags = append(metric.Tags, instance.Tags...)
-						metric.Tags = append(metric.Tags, fmt.Sprintf("%s:%s", "region", instance.Region))
-					}
-				}
-				metric.Tags = append(metric.Tags, fmt.Sprintf("%s:%s", types.StatsdSourceID, hostname))
+			if !triedToGetTags {
```

Comment:
> This nil thing is confusing a bit (because it is unusual and we thing about it like about NULL) but I'm getting used to it...

Yes it's confusing, especially that it is not a proper return value for a `struct` but it is for the `pointer to a struct`...